backend/dataall/modules/redshift_datasets_shares/cdk/pivot_role_redshift_data_sharing_policy.py

```diff
@@ -10,12 +15,35 @@ class RedshiftDataSharingPivotRole(PivotRoleStatementSet):
     def get_statements(self):
         statements = [
             iam.PolicyStatement(
-                sid='RedshiftReadShareResources',  # These permissions can only be applied to *
+                sid='RedshiftReadShareResources',
                 effect=iam.Effect.ALLOW,
-                actions=['redshift-data:GetStatementResult'],
+                actions=[
+                    'redshift-data:GetStatementResult',  # It can only be applied to *
+                    'redshift:AssociateDataShareConsumer',  # From consumer we can accept any share
+                    'redshift:DescribeDataShares',  # Read
+                ],
                 resources=[
                     '*',
                 ],
-            )
+            ),
         ]
-        return statements
+        additional_statements = []
+        engine = db.get_engine(envname=os.environ.get('envname', 'local'))
+        with engine.scoped_session() as session:
+            connections = RedshiftConnectionRepository.list_environment_redshift_connections(
+                session, environment_uri=self.environmentUri
+            )
+            if connections:
+                source_datashare_arns = [
+                    f'arn:aws:redshift:{self.region}:{self.account}:datashare:{conn.nameSpaceId}/*'
```

Comment:
do we also need for `redshift-serverless`?

Reply:
nevermind - seems datashares always use `redshift` and from testing look like this works as is